cpp/cmake-library/build.gradle

```diff
@@ -0,0 +1,53 @@
+plugins {
+    id 'cpp-application'
+    id 'xcode'
+}
+
+configurations {
+    cmake
+}
+
+def cmakeLibraryPackage = file('library/libraries-0.1.1-Darwin.zip')
```

Comment:
i think it's important that we show how to use variant dependency management so that the consuming project can't tell the difference between a Gradle built c++ library and a cmake built library. We should use dependency management as the interface between the different kinds of producers, so that every kind of producer ends up producing something that can be used by every kind of consumer.

So, I would structure this as a build with 2 projects:

- one project, "app", applies the `cpp-application` plugin and simply depends on the other project.
- the other project, "lib" has a Gradle build script that calls cmake to build the binaries and some configurations that expose the result. This would evolve into the cmake wrapper plugin.

Reply:
That makes sense.  I actually started doing something like that and then wasn't sure how much we wanted Gradle to bleed into the "legacy" project.  There may be another use case here where the legacy project is more of a black box and we just get some sort of idiomatic package out of it, but to your point, this is probably less interesting.  I'll migrate it towards something where the interface is driven by the dependency mgmt engine.

Reply:
Ok, this has been converted so that the cmake project is a Gradle build that wraps cmake and exports the artifacts as if they were built by Gradle.